mount/mountinfo_linux_test.go

```diff
@@ -490,3 +493,48 @@ func TestParseFedoraMountinfoFields(t *testing.T) {
 		t.Fatalf("expected %#v, got %#v", mi, infos[0])
 	}
 }
+
+func TestParseMountinfoWithSpaces(t *testing.T) {
+	r := bytes.NewBuffer([]byte(mountInfoWithSpaces))
+	infos, err := parseInfoFile(r)
+	if err != nil {
+		t.Fatal(err)
+	}
+	expected := []Info{
+		{
+			ID:         486,
+			Parent:     28,
+			Major:      252,
+			Minor:      1,
+			Root:       "/",
+			Mountpoint: "/mnt/foo bar",
+			Options:    "rw,relatime",
+			Optional:   "shared:243",
+			FSType:     "ext4",
+			Source:     "/dev/vda1",
+			VFSOptions: "rw,data=ordered",
+		},
+		{
+			ID:         31,
+			Parent:     21,
+			Major:      0,
+			Minor:      23,
+			Root:       "/",
+			Mountpoint: "/DATA/foo_bla_bla",
+			Options:    "rw,relatime",
+			Optional:   "",
+			FSType:     "cifs",
+			Source:     `//foo/BLA\040BLA\040BLA/`,
+			VFSOptions: `rw,sec=ntlm,cache=loose,unc=\\foo\BLA`,
```

Comment:
Note that this is truncating all options after the space; this is documented in `parseInfoFile` as expected, but I'm thinking of doing a follow-up to preserve the information that was removed (this might need some discussion, so I'm keeping that separate)